File: Redcoder/Python/88.py
"""
ランレングス圧縮を用いた圧縮方法
書き方は最適化はわからないが行っていることは
AAABBBBBCC
の文字列があった場合に
A:3 B:5 C:2
のようにまとめた形で表現する圧縮方法
"""

N = int(input())
box = []
a = int(input())
box.append([a,1])
for i in range(N-1):
    a = int(input())
    if i % 2 == 1:
        if box[-1][0] == a:
            box[-1][1] += 1
        else:
            box.append([a,1])
        continue
    if box[-1][0] == a:
        box[-1][1] += 1
    else:
        box[-1][0] += 1
        box[-1][0] %= 2
        box[-1][1] += 1
        if len(box) >=2 and box[-1][0] == box[-2][0]:
            b,c = box.pop()
            box[-1][1] += c
ans = 0
for i in range(len(box)):
    if box[i][0] == 0:
        ans += box[i][1]
print(ans)

File: Redcoder/C++/88.cpp
#include <bits/stdc++.h>
#define rep(i, nn, n) for (int i = nn; i < (n); ++i)
#define rrep(i, nn, n) for (int i = nn; i > (n); --i)
using namespace std;
using ll = long long;
using P = pair<int, int>;
#define chmin(x, y) x = min(x, y)
#define chmax(x, y) x = max(x, y)

int main()
{
    int n;
    cin >> n;
    vector<P> box;
    int a;
    cin >> a;
    box.push_back({a, 1});
    rep(i, 0, n - 1)
    {
        int num;
        cin >> num;
        if (i % 2 == 1)
        {
            if (box[box.size() - 1].first == num)
                ++box[box.size() - 1].second;
            else
                box.push_back({num, 1});
        }
        else
        {
            if (box[box.size() - 1].first == num)
                ++box[box.size() - 1].second;
            else
            {
                int count = 1 + box[box.size() - 1].second;
                box.pop_back();
                if (box.size() == 0)
                    box.push_back({num, count});
                else
                    box[box.size() - 1].second += count;
            }
        }
    }
    int ans = 0;
    rep(i, 0, box.size())
    {
        if (box[i].first == 0)
            ans += box[i].second;
    }

    cout << ans << endl;
    return 0;
}